app/hydrator/services/hydrator-upgrade-service.js

```diff
@@ -174,6 +174,12 @@ class HydratorUpgradeService {
 
   _checkErrorStages(stages, pluginsMap) {
     let transformedStages = [];
+    if (!stages || !stages.forEach) {
```

Comment:
is this meant to check if its a immutable map as well as an array?

Reply:
This is meant to guard against unknown types since I only have a single failure example and I don't know what's possible. I could clarify the comment if you like!

Reply:
yeah can we add a comment then? or is that linked in the jira?